cpu/stm32/stmclk/stmclk_gx.c

```diff
@@ -191,73 +220,83 @@ void stmclk_init_sysclk(void)
+        /* configure HSISYS divider, only available on G0 */
+        RCC->CR |= CLOCK_HSI_DIV;
+        while (!(RCC->CR & RCC_CR_HSIRDY)) {}
+    }
 #endif
 
-#if CONFIG_USE_CLOCK_HSI && defined(CPU_FAM_STM32G0)
-    /* configure HSISYS divider, only available on G0 */
-    RCC->CR |= CLOCK_HSI_DIV;
-    while (!(RCC->CR & RCC_CR_HSIRDY)) {}
-
-#elif CONFIG_USE_CLOCK_HSE
-    /* if configured, we need to enable the HSE clock now */
-    RCC->CR |= RCC_CR_HSEON;
-    while (!(RCC->CR & RCC_CR_HSERDY)) {}
-
+    /* Enable HSE if required */
+    if (IS_ACTIVE(CLOCK_ENABLE_HSE)) {
+        RCC->CR |= RCC_CR_HSEON;
+        while (!(RCC->CR & RCC_CR_HSERDY)) {}
+    }
+
+    /* Enable PLL if required */
+    if (IS_ACTIVE(CLOCK_ENABLE_PLL)) {
+        RCC->PLLCFGR = (PLL_SRC | PLL_M | PLL_N | PLL_R | RCC_PLLCFGR_PLLREN);
+        RCC->CR |= RCC_CR_PLLON;
+        while (!(RCC->CR & RCC_CR_PLLRDY)) {}
+    }
+
+    /* Configure SYSCLK */
+    if (IS_ACTIVE(CONFIG_USE_CLOCK_HSE)) {
 #if defined(CPU_FAM_STM32G0)
-    RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV);
+        RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV);
 #elif defined(CPU_FAM_STM32G4)
-    RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV | CLOCK_APB2_DIV);
-#endif
-    while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSE) {}
-
-#elif CONFIG_USE_CLOCK_PLL
-#if CONFIG_BOARD_HAS_HSE
-    /* if configured, we need to enable the HSE clock now */
-    RCC->CR |= RCC_CR_HSEON;
-    while (!(RCC->CR & RCC_CR_HSERDY)) {}
+        RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV | CLOCK_APB2_DIV);
 #endif
-    /* now we can safely configure and start the PLL */
-    RCC->PLLCFGR = (PLL_SRC | PLL_M | PLL_N | PLL_R | RCC_PLLCFGR_PLLREN);
-    RCC->CR |= RCC_CR_PLLON;
-    while (!(RCC->CR & RCC_CR_PLLRDY)) {}
-
-#if CLOCK_AHB > MHZ(80)
-    /* Divide HCLK by before enabling the PLL */
-    RCC->CFGR |= RCC_CFGR_HPRE_DIV2;
+        RCC->CFGR |= RCC_CFGR_SW_HSE;
+        while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSE) {}
+    }
+    else if (IS_ACTIVE(CONFIG_USE_CLOCK_PLL)) {
+#if defined(CPU_FAM_STM32G4)
+        if (CLOCK_AHB > MHZ(80)) {
+            /* Divide HCLK by 2 before enabling the PLL */
+            RCC->CFGR |= RCC_CFGR_HPRE_DIV2;
+        }
 #endif
 
-    /* now that the PLL is running, we use it as system clock */
-    RCC->CFGR |= RCC_CFGR_SW_PLL;
-    while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_PLL) {}
-
-#if CLOCK_AHB > MHZ(80)
-    /* Wait 1us before switching back to full speed */
-    /* Use volatile to prevent the compiler from optimizing the loop */
-    volatile uint8_t count = CLOCK_CORECLOCK / MHZ(1);
-    while (count--) {}
-    RCC->CFGR &= ~RCC_CFGR_HPRE_DIV2;
-#endif
+        /* now that the PLL is running, we use it as system clock */
+        RCC->CFGR |= RCC_CFGR_SW_PLL;
+        while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_PLL) {}
 
+#if defined(CPU_FAM_STM32G4)
```

Comment:
```suggestion
#if defined(RCC_CFGR_HPRE_DIV2)
```

would that also make sens?

Reply:
Same answer as https://github.com/RIOT-OS/RIOT/pull/15036/files#r508019638.

Removing preprocessor checks that are family specific makes sense in files that are shared by a lot more families (like I2C, UART, etc) because they are quite different and some defines/features are slightly different. In stmclk, this is different because the implementation are grouped by very similar families (f0/f1/f3, f2/f4/f7, g0/g4, l4/wb), so not more than 3 different families. In this case, it's cleaner and more explicit to use preprocessor conditionals based on the family and not on the existence of an obscure define.